src/folders.ts

```diff
@@ -1,31 +1,101 @@
+import { PresetsController } from '@cmt/presetsController';
+import * as preset from '@cmt/preset';
 
 nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
 const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 
 export class CMakeToolsFolder {
-  private constructor(readonly cmakeTools: CMakeTools, readonly kitsController: KitsController) { }
+  private _wasUsingCMakePresets: boolean | undefined;
+
+  private readonly _onUseCMakePresetsChangedEmitter = new vscode.EventEmitter<boolean>();
+
+  private constructor(readonly cmakeTools: CMakeTools,
+                      readonly kitsController: KitsController,
+                      readonly presetsController: PresetsController) {
+    const useCMakePresetsChangedListener = async () => {
+      const usingCMakePresets = this.useCMakePresets;
+      if (usingCMakePresets !== this._wasUsingCMakePresets) {
+        this._wasUsingCMakePresets = usingCMakePresets;
+        setContextValue('useCMakePresets', usingCMakePresets);
+        await cmakeTools.setUseCMakePresets(usingCMakePresets);
+        await CMakeToolsFolder.initializeKitOrPresetsInCmt(this);
+        this._onUseCMakePresetsChangedEmitter.fire(usingCMakePresets);
+      }
+    };
+    cmakeTools.workspaceContext.config.onChange('useCMakePresets', useCMakePresetsChangedListener);
+    preset.onPresetsChanged(useCMakePresetsChangedListener);
+    preset.onUserPresetsChanged(useCMakePresetsChangedListener);
+  }
 
   static async init(cmakeTools: CMakeTools) {
-    return new CMakeToolsFolder(cmakeTools, await KitsController.init(cmakeTools));
+    const kitsController = await KitsController.init(cmakeTools);
+    const cmtFolder = new CMakeToolsFolder(cmakeTools, kitsController, await PresetsController.init(cmakeTools, kitsController));
+    const usingCMakePresets = cmtFolder.useCMakePresets;
+    cmtFolder._wasUsingCMakePresets = usingCMakePresets;
+    setContextValue('useCMakePresets', usingCMakePresets);
+    await cmakeTools.setUseCMakePresets(usingCMakePresets);
+    await CMakeToolsFolder.initializeKitOrPresetsInCmt(cmtFolder);
+    return cmtFolder;
   }
 
   get folder() { return this.cmakeTools.folder; }
 
+  // Go through the decision tree here since there would be dependency issues if we do this in config.ts
+  get useCMakePresets(): boolean {
+    if (this.cmakeTools.workspaceContext.config.useCMakePresets === 'automatic') {
+      // TODO (P1): check if configured with kits + vars
+      // // Always check if configured before since the state could be reset
+      // const state = this.cmakeTools.workspaceContext.state;
+      // const configuredWithKitsVars = !!(state.activeKitName || state.activeVariantSettings?.size);
+      // return !configuredWithKitsVars || (configuredWithKitsVars && (this.presetsController.cmakePresetsExist || this.presetsController.cmakeUserPresetsExist));
+      return this.presetsController.presetsFileExist;
```

Comment:
Double commented comment, commented typescript.... maybe you forgot to delete a block of source code here?

Reply:
Commented out part is P1, which is what this code will look like later.